docs/using.md

```diff
@@ -238,6 +238,113 @@ Use the following example to understand the syntax and strucutre of a Git Resour
       serviceAccount: build-bot
     ```
 
+### Cluster Resource
+
+Cluster Resource represents a Kubernetes cluster other than the current cluster the pipleine 
+CRD is running on. A common use case for this resource is to deploy your application/function 
+on different clusters. 
+
+The resource will use the provided parameters to create a [kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) file 
+that can be used by other steps in the pipeline task to access the target cluster. The kubeconfig will be 
+placed in `/workspace/<your-cluster-name>/kubeconfig` on your task container
+
+The Cluster resource has the following parameters:
+ * URL (required): Host url of the master node         
+ * ClusterName (required): name given to cluster, will be used in the kubeconfig and also as part of the path to the kubeconfig file 
```

Comment:
is this different from the name of the resource? would it make sense to use the name of the resource for the cluster name as well? (I'd expect the kubeconfig file to be created at `/workspace/<your-cluster-resource-name>/kubeconfig`

Reply:
makes sense, will use Name